webcompat/helpers.py

```diff
@@ -408,6 +408,35 @@ def wrapped_func(*args, **kwargs):
     return wrapped_func
 
 
+def mockable_response_direct(func, path, get_args=None):
+    """Mock out API reponses with a decorator.
+
+    This allows us to send back fixture files when in TESTING mode, rather
+    than making API requests over the network. See /api/endponts.py
+    for usage.
+    """
+    @wraps(func)
+    def wrapped_func(*args, **kwargs):
+        if app.config['TESTING']:
+            full_path = path.format(*args)
+            if get_args:
+                # if there are GET args, encode them as a hash so we can
+                # have different fixture files for different response states
+                checksum = hashlib.md5(json.dumps(get_args)).hexdigest()
+                file_path = FIXTURES_PATH + full_path + "." + checksum
+            else:
+                file_path = FIXTURES_PATH + full_path
+            if not os.path.exists(file_path + '.json'):
+                print('Expected fixture file: ' + file_path + '.json')
+                return ('', 404)
+            else:
+                with open(file_path + '.json', 'r') as f:
+                    data = f.read()
+                    return (data, 200, get_fixture_headers(data))
+        return func(*args, **kwargs)
+    return wrapped_func
```

Comment:
I had to add this function and keep the old `mockable_response` because there are still parts of the code rendered in js but I needed a new function to deal with the python rendered part.

The idea here is that `wrapped_func` now receives as an argument the path of the request that was made, that's why I added a string on the function on the `endpoints.py` ` proxy_comments_direct = mockable_response_direct(proxy_comments_impl, '/api/issues/{0}/comments')`. With that string I'm able to make the fake requests and get the correct mocks. :)

Reply:
ok basically if I understand it's code to make it work, but not code necessary if we revise the tests.

Reply:
Yes, exactly. I know it's ugly. 

To make it pretty we would have to change how the test mode works. Currently it's completely dependent of `request.args` but it can't be at all because since we're not making real requests this attribute is useless.